packages/components/src/internal/components/search/SampleFinderSavedViewsMenu.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { FC, memo, useState, useEffect, useMemo } from 'react';
+import { Button, DropdownButton, MenuItem } from 'react-bootstrap';
+
+import { LoadingSpinner } from '../base/LoadingSpinner';
+import { SubMenuItem } from '../menus/SubMenuItem';
+
+import { ComponentsAPIWrapper, getDefaultAPIWrapper } from '../../APIWrapper';
+
+import { FinderReport } from './models';
+
+interface Props {
+    api?: ComponentsAPIWrapper;
+    loadSearch: (view: FinderReport) => any;
+    saveSearch: (saveCurrentName?: boolean) => any;
+    manageSearches: () => any;
+    currentView?: FinderReport;
+    hasUnsavedChanges?: boolean;
+    sessionViewName?: string;
+    key: any;
+}
+
+export const SampleFinderSavedViewsMenu: FC<Props> = memo(props => {
+    const { api, loadSearch, manageSearches, saveSearch, currentView, hasUnsavedChanges, sessionViewName, key } = props;
+
+    const [savedSearches, setSavedSearches] = useState<FinderReport[]>(undefined);
+
+    useEffect(() => {
+        (async () => {
+            const views = await api.samples.loadFinderSearches();
+            setSavedSearches(views);
+        })();
+    }, [key]);
+
+    const hasSavedView = savedSearches?.length > 0;
+
+    const menuTitle = useMemo(() => {
+        if (!currentView?.reportId) return 'Saved Searches';
+        return (
+            <>
+                {hasUnsavedChanges && <span className="alert-info finder-view-edit-alert">EDITED</span>}
+                {currentView.reportName}
+            </>
+        );
+    }, [currentView, hasUnsavedChanges]);
+
+    return (
+        <>
+            <DropdownButton id="samplefinder-savedsearch-menu" title={menuTitle} className="button-right-spacing">
+                {sessionViewName && (
+                    <>
+                        <SubMenuItem text="MOST RECENT SEARCH" inline>
```

Comment:
instead of having these all caps, should we add a css style for transform to uppercase? I think Andrew mentioned we'll be making that change for all of our submenu titles.